tangelo/algorithms/classical/fci_solver.py

```diff
@@ -131,3 +144,170 @@ def get_rdm(self):
                 one_rdm, two_rdm = self.cisolver.make_rdm12(self.ci, self.norb, (self.n_alpha, self.n_beta))
 
         return one_rdm, two_rdm
+
+
+class FCISolverPsi4(ElectronicStructureSolver):
+    """ Uses the Full CI method to solve the electronic structure problem,
+    through Psi4.
+
+    Args:
+        molecule (SecondQuantizedMolecule): The molecule to simulate.
+
+    Attributes:
+        ciwfn (psi4.core.CIWavefunction): The CI wavefunction (float64).
+        backend (psi4): The psi4 module
+        molecule (SecondQuantizedMolecule): The molecule with symmetry=False
+    """
+
+    def __init__(self, molecule: SecondQuantizedMolecule):
+        if not is_package_installed("psi4"):
+            raise ModuleNotFoundError(f"Using {self.__class__.__name__} requires the installation of the Psi4 package")
+
+        if molecule.uhf:
+            raise NotImplementedError(f"SecondQuantizedMolecule that use UHF are not currently supported in {self.__class__.__name__}. Use CCSDSolver")
+
+        import psi4
+        self.backend = psi4
+        self.backend.core.clean_options()
+        self.ciwfn = None
+        self.any_close = False
+        if isinstance(molecule.solver, IntegralSolverPsi4) and not molecule.symmetry:
+            self.molecule = molecule
+        else:
+            print(molecule.mo_energies)
+
+            for i, val in enumerate(molecule.mo_energies[:-1]):
```

Comment:
Could be done without a python loop by rounding the floats and getting the unique elements, then compare the array shapes.
```python
unique_arr = np.unique(arr.round(decimals=tolerance))
self.any_close = (unique_arr.shape != arr.shape)
```
(not tested, at your own risk) Another option would be to look at the count array returned by the `np.unique` function.